list-doi-requests/src/test/java/no/unit/nva/doirequest/list/ListDoiRequestsHandlerTest.java

```diff
@@ -181,47 +195,130 @@ public void listDoiRequestsReturnsEmptyListForUserWhenUserHasNoValidRole()
         assertThat(responseBody, is(emptyCollectionOf(Publication.class)));
     }
 
+    @Test
+    public void listDoiRequestsForUserReturnsDtosWithMessagesIncludedWhenThereAreDoiRequestMessagesForTheDoiRequest()
+        throws ApiGatewayException, IOException {
+        List<Publication> publications = createPublishedPublicationsOfSameOwner();
+        createDoiRequests(publications);
+        final var doiRequestMessages = creteDoiRequestMessagesForPublications(publications);
+
+        URI commonPublisherId = publications.get(FIRST_ELEMENT).getPublisher().getId();
+        String commonOwner = publications.get(FIRST_ELEMENT).getOwner();
+        List<String> actualMessages = sendRequestAndReadMessages(commonOwner, commonPublisherId, CREATOR_ROLE);
+
+        String[] expectedMessages = extractExpectedMessageTexts(doiRequestMessages);
+        assertThat(actualMessages, containsInAnyOrder(expectedMessages));
+    }
+
+    @Test
+    public void listDoiRequestsForCuratorReturnsDtosWithMessagesIncludedWhenThereAreDoiRequestMessagesForTheDoiRequest()
+        throws ApiGatewayException, IOException {
+        var publications = createPublishedPublicationsOfSamePublisherButDifferentOwner();
+        createDoiRequests(publications);
+        final var doiRequestMessages = creteDoiRequestMessagesForPublications(publications);
+
+        URI commonPublisherId = publications.get(0).getPublisher().getId();
+        List<String> actualMessages = sendRequestAndReadMessages(SOME_CURATOR, commonPublisherId, CURATOR_ROLE);
+
+        String[] expectedMessages = extractExpectedMessageTexts(doiRequestMessages);
+        assertThat(actualMessages, containsInAnyOrder(expectedMessages));
```

Comment:
Im a little confused when you compare a List against an Array but maybe its needed that way, i'm not sure.

Reply:
Unfortunately this is the way hamcrest works. If one adds a list as an argument in the containsInAnyOrder, then hamcrest will think that you have a List of Lists.